extensions/binding/org.eclipse.smarthome.binding.mqttbroker/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,120 @@
+				<label>Last will topic</label>
+				<description>The last will topic. Defaults to empty and therefore disables the last will.</description>
+			</parameter>
+
+			<parameter name="lastwill_qos" type="integer">
+				<label>Last will QoS</label>
+				<description>The quality of service parameter of the last will.</description>
+				<options>
+					<option value="0">0</option>
+					<option value="1">1</option>
+					<option value="2">2</option>
+				</options>
+				<default>0</default>
+			</parameter>
+
+			<parameter name="username" type="text">
+				<label>username</label>
+				<description>The Mqtt username (since Mqtt 3.1)</description>
+			</parameter>
+			<parameter name="password" type="text">
+				<label>password</label>
+				<description>The Mqtt password (since Mqtt 3.1)</description>
+			</parameter>
+			<parameter name="certificatepin" type="boolean">
+				<label>Certificate pinning</label>
+				<description>If this and SSL is set: After the next connection has been successfully established, the certificate is pinned. The connection will be refused if another certificate is used. Clear **certificate** to allow a new certificate for the next connection attempt. This option can increase security.</description>
+				<default>true</default>
+			</parameter>
+			<parameter name="certificate" type="text">
+				<label>certificate</label>
+				<description>The certificate hash. If **certificatepin** is set this hash is used to verify the connection. Clear to allow a new certificate pinning on the next connection attempt. If empty will be filled automatically by the next successful connection. An example input would be `CERTSHA256:83F9171E06A313118889F7D79302BD1B7A2042EE0CFD029ABF8DD06FFA6CD9D3` or `CERTSHA256:83:F9:17:1E:06:A3:13:11:88:89:F7:D7:93:02:BD:1B:7A:20:42:EE:0C:FD:02:9A:BF:8D:D0:6F:FA:6C:D9:D3`.</description>
+			</parameter>
+			<parameter name="publickeypin" type="boolean">
+				<label>Public key pinning</label>
+				<description>If this and SSL is set: After the next connection has been successfully established, the public key of the broker is pinned. The connection will be refused if another public key is used. Clear **publickey** to allow a new public key for the next connection attempt. This option can increase security.</description>
+				<default>true</default>
+			</parameter>
+			<parameter name="publickey" type="text">
+				<label>publickey</label>
+				<description>The public key hash. If **publickeypin** is set this hash is used to verify the connection. Clear to allow a new public key pinning on the next connection attempt. If empty will be filled automatically by the next successful connection. An example input would be `CERTSHA256:83F9171E06A313118889F7D79302BD1B7A2042EE0CFD029ABF8DD06FFA6CD9D3` or `CERTSHA256:83:F9:17:1E:06:A3:13:11:88:89:F7:D7:93:02:BD:1B:7A:20:42:EE:0C:FD:02:9A:BF:8D:D0:6F:FA:6C:D9:D3`</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<thing-type id="TextualBroker">
```

Comment:
Impossible. The architecture (the majority is inherited by the old MqttService implementation) is as follows:

* The MqttService, as a shared pool of connections. 
* Connections can be added by using the mqtt.cfg file. Those entries go into OSGi ConfigAdmin and are published to the MqttService. Those connections are immutable.
* Connections can be added by using this binding. The connections are not stored in ConfigAdmin, but in the jsondb as Things instead. This allows with means that already exists (PaperUI, the REST interface) to provide a GUI for configuring MQTT connections.
* The GUI though can not change immutable connections, but it can list them for the user for easier debugging. Those connections are presented as "TextualBroker". Those Things provide read-only properties of the connection, for example the error message and connection status.

Reply:
Ok, I see a bit clearer now. The wording misled me because "textual" usually refers to DSL defined things, i.e. those provided through a *.things file, which is not what we talk about here.

> Connections can be added by using the mqtt.cfg file.

... and through other means that either provide configurations or use the MqttService programatically. Therefore "textual" is definitely a wrong term here. I'd rather suggest "SystemBroker" or similar to make it clear that this is a broker that is defined centrally in the system.

> Connections can be added by using this binding. 

Just to talk about the same: "Connections" are bridges of "Broker" type (the one above) and they are NOT added to the MqttService, but exist independently of it and can be used by specific MQTT-Thing bindings.

> The GUI though can not change immutable connections, but it can list them for the user for easier debugging. 

Not only for debugging, but mainly to allow users of an MQTT-Thing binding to select those as a bridge. 
So yes, you are correct that we need a separate bridge-type for those. And yes, for this bridge-type, it is ok to have relevant information added as properties. But you NEED some configuration parameters for this bridge-type, which let's the handler identify how to establish the connection. In this case, this is probably simply the "brokerName", which is used within MqttService for identifying an existing connection, right?

Reply:
> Just to talk about the same: "Connections" are bridges of "Broker" type (the one above) and they are NOT added to the MqttService, but exist independently of it and can be used by specific MQTT-Thing bindings.
They are added to MqttService, with the idea that MqttService is the shared pool and the mqtt.cfg file is the legacy method of adding connections. But they can be distinguished, the mqtt.cfg connections have a flag set.

> But you NEED some configuration parameters for this bridge-type, which let's the handler identify how to establish the connection.
This binding and the handlers are not responsible for those system connections, they are started as soon as the configuration is read in modified(). This is mostly, because we wanted to stay backwards compatible and the old mqtt.cfg connections are auto-started.